src/transformStateWithClones.js

```diff
@@ -1,13 +1,42 @@
 'use strict';
 
 /**
- * @param {Object} state
+ * @param {Object} stateCopy
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  let stateCopy = { ...state };
+
+  for (const object of actions) {
+    if (object.type === 'addProperties') {
+      const extraData = object.extraData;
+
+      Object.assign(stateCopy, extraData);
+
+      resultArray.push({ ...stateCopy });
+    }
+
+    if (object.type === 'removeProperties') {
+      const keyNamesArray = object.keysToRemove;
+
+      for (const keyName of keyNamesArray) {
+        delete stateCopy[keyName];
+      }
+      resultArray.push(Object.assign({}, stateCopy));
+    }
+
+    if (object.type === 'clear') {
+      const empty = {};
+
+      stateCopy = empty;
+      resultArray.push({ ...stateCopy });
```

Comment:
The same issue regarding constants
```suggestion
      resultArray.push({});
```

Reply:
![image](https://github.com/mate-academy/js_stateful-clones/assets/146844312/1a588b66-430f-4eea-955c-f81a31f19194)
it doesnt work((